core/trino-main/src/main/java/io/trino/sql/gen/RowConstructorCodeGenerator.java

```diff
@@ -88,4 +97,38 @@ public BytecodeNode generateExpression(BytecodeGeneratorContext context)
         block.append(context.wasNull().set(constantFalse()));
         return block;
     }
+
+    private BytecodeNode generateExpressionForLargeRows(BytecodeGeneratorContext context)
+    {
+        BytecodeBlock block = new BytecodeBlock().setDescription("Constructor for " + rowType);
```

Comment:
Can you add a comment on how this is different from the above code.  Specifically touch on where we get less byte code and why the performance is not as good as above (otherwise we would use the same code for both).